File: languages/python/bloock/entity/identity/integer_attribute_descriptor.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.attribute_descriptor import AttributeDescriptor


class IntegerAttributeDescriptor(AttributeDescriptor):
    """
    Represents a descriptor for an attribute with an integer value.
    """
    @staticmethod
    def from_proto(a: proto.IntegerAttributeDefinition) -> IntegerAttributeDescriptor:
        return IntegerAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
        )

    def to_proto(self) -> proto.IntegerAttributeDefinition:
        return proto.IntegerAttributeDefinition(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
        )

File: languages/php/Bloock/Entity/Identity/IntegerAttributeDescriptor.php
<?php

namespace Bloock\Entity\Identity;

use Bloock\IntegerAttributeDefinition;

/**
 * Represents a descriptor for an attribute with an integer value.
 */
class IntegerAttributeDescriptor extends AttributeDescriptor
{
    public static function fromProto(IntegerAttributeDefinition $res): IntegerAttributeDescriptor
    {
        return new IntegerAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired());
    }

    public function toProto(): IntegerAttributeDefinition
    {
        $p = new IntegerAttributeDefinition();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        return $p;
    }
}